controllers/flowcollector_controller.go

```diff
@@ -103,6 +104,15 @@ func (r *FlowCollectorReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 	previousNamespace := desired.Status.Namespace
 	reconcilersInfo := r.newCommonInfo(ctx, desired, ns, previousNamespace, func(b bool) { didChange = b }, func(b bool) { isInProgress = b })
 
+	// obtain default cluster ID
+	cversion := &configv1.ClusterVersion{}
+	key := client.ObjectKey{Name: "version"}
+	if err := r.Client.Get(ctx, key, cversion); err != nil {
+		log.Error(err, "unable to obtain cluster ID")
+	} else {
+		constants.DefaultClusterID = cversion.Spec.ClusterID
+	}
```

Comment:
This block should be run only once, and only for openshift. To do so you can wrap it in a block  `if r.permissions.Vendor(ctx) == discover.VendorOpenShift && constants.DefaultClusterID == "" {`

Reply:
Also it is unexpected to update a "constant". I would move `DefaultClusterID` somewhere else. For instance, it could be part of the `FlowCollectorReconciler` struct, and passed down to reconcilers via `newCommonInfo`; or if you prefer to keep a global access, maybe create a new file `globals.go` in package `reconcilers` ?

Reply:
In order to pass down the clusterName through the series of function calls it would have required many changes and would have made the code ugly. I therefore made it into a global, as suggested.

Reply:
That's ok
The "CommonInfo" is already done in that purpose, to be passed to the reconcilers, so you wouldn't have to change all the function signatures, just adding it to a struct that is already spread across reconcilers.
But that's ok with the global approach - as long as not everything ends up like that, which would denote a lack of modularity and makes sometimes testing more difficult